### ISSUE TITLE
Strange graph revision when using filtering feature

### ISSUE THREAD

**Issue by pmiossec** [gitextensions/gitextensions#5454]
**Do you want to request a *feature* or report a *bug*?**
A bug (maybe). At best a strange behavior.

**What is the current behavior?**

A lot of branches in gray are displayed and I found that hard to understand and misleading.
And also each commit is displayed (except ones with direct parent/child also in the filter result) in its own gray branch even if they are ancestor of the current branch (which is displayed in pink)

![image](https://user-images.githubusercontent.com/460196/45739561-95d4d300-bbf3-11e8-82ca-f2cdcf9f0098.png)

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

- "Filter" the revision grid with a text

**What is the expected behavior?**

2 possibilities...

Either we could get a good graph where all the commits contained in the current branch (in pink) is displayed in the same pink branch and only the commits not contained in the current branch are displayed in gray branches

Or we hide the graph...

**Environment you encounter the issue:**

- GitExtensions version: all versions for all long I use GitExtensions ;)
- GIT version: 2.18
- OS version: win10

**Did this work in previous version of GitExtensions (which)?**
No.

**Comment by RussKie** [comment]
Can it be related to #5450 in any way?

**Comment by drewnoakes** [comment]
> Can it be related to #5450 in any way?

I don't believe so. I think I've seen this for a while, and @pmiossec suggests the same.

**Comment by pmiossec** [comment]
>Can it be related to #5450 in any way?

I really don't think so (even if I have not tested #5450).
This issue make me remember this long time strange behavior, that's why I thought about creating the issue.

Even if I no more have a lot of free time, perhaps I will try to have a look.
But I don't know the code managing the graph.
If someone has some quick advises on how it works....

**Comment by gerhardol** [comment]
It is not so easy as it seems to collapse a graph hiding nodes.
The commits still need to save the real parents and the structure does not have a way to have a separate "graph parent". So I see no simple solution here to show the graph.
On the other hand, there can be scenarios where you have an unbroken list of parents so to always drop the graphs when filtering will limit the functionality.

![image](https://user-images.githubusercontent.com/6248932/45785164-9dd05980-bc6b-11e8-8b89-7cb1c01f3acf.png)

![image](https://user-images.githubusercontent.com/6248932/45785200-b8a2ce00-bc6b-11e8-913b-c755925a6030.png)

I say that we do nothing for now.

**Comment by vbjay** [comment]
I agree. I think the user knows they are filtering.

**Comment by pmiossec** [comment]
>I agree. I think the user knows they are filtering.

That's not the problem. The problem is to know which commits are contained in the current branch.
And in this sense, the graph is at best not understandable and at worst, misleading.

So the graph is in this case of no help. In more than 6 years, I'm not sure I used it one time because of that (even if I tried quite some times). And when you scroll, it became even worst... 

So, my point of view is, if we can't fix it, we surely should to hide it because "user knows they are filtering".

And perhaps find another way to highlight commits that are contained in the current branch.

**Comment by gerhardol** [comment]
Low prio for me but some possible solutions:
 * Make sure collapsing handles removal of commits including head, merges, etc
 * Replace with some marking/collapsing in the graph instead

**Comment by pmiossec** [comment]
There is also a good description of the problem in #6614
